crates/cxx-qt-gen/src/generator/cpp/signal.rs

```diff
@@ -9,11 +9,57 @@ use crate::{
         naming::{qobject::QObjectName, signals::QSignalName},
         utils::cpp::syn_type_to_cpp_type,
     },
-    parser::{mappings::ParsedCxxMappings, signals::ParsedSignal},
+    parser::{
+        mappings::ParsedCxxMappings, parameter::ParsedFunctionParameter, signals::ParsedSignal,
+    },
 };
 use indoc::formatdoc;
 use syn::Result;
 
+/// Combined output of possible parameter lines to be used
+struct Parameters {
+    types_closure: String,
+    types_signal: String,
+    values_closure: String,
+}
+
+/// Representation of the self pair
+///
+/// This allows for *this being passed or a parameter in the arguments
+struct SelfValue<'a> {
+    ident: &'a str,
+    ty: &'a str,
+}
+
+/// From given parameters, mappings, and self value constructor the combined parameter lines
+fn parameter_types_and_values(
+    parameters: &Vec<ParsedFunctionParameter>,
```

Comment:
done, although feels like this could be done in a lot of other places too?

Should we add a "good first issue" to go through and check for `&Vec<T>` ->`&[T]` ?